jwst/lib/basic_utils.py

```diff
@@ -1,5 +1,31 @@
 """General utility objects"""
 
+from stdatamodels.jwst.datamodels import dqflags
+import numpy as np
+
+
+def set_nans_to_donotuse(data, dq):
+    """Set all NaN values in the data that have an even value to
```

Comment:
I know I'm late to this party (after the PR has already been merged), but this description doesn't really seem to match the code. The code is simply adding (bitwise "or"ing) the DO_NOT_USE flag to all pixels that are set to NaN in the data array. It's not checking for anything like even or odd DQ values.